vscode/src/client/analyzerClient.ts

```diff
@@ -5,47 +5,81 @@ import * as fs from "fs";
 import { Incident, RuleSet } from "@editor-extensions/shared";
 
 import path from "path";
+import { ExtensionState } from "../extensionState";
 
 export class AnalyzerClient {
   private config: vscode.WorkspaceConfiguration | null = null;
   private extContext: vscode.ExtensionContext | null = null;
   private analyzerServer: ChildProcessWithoutNullStreams | null = null;
   private outputChannel: vscode.OutputChannel;
-  // private rpcConnection: rpc.MessageConnection | null = null;
   private requestId: number = 1;
+  private state: ExtensionState;
 
-  constructor(context: vscode.ExtensionContext) {
-    this.extContext = context;
+  constructor(state: ExtensionState) {
     this.outputChannel = vscode.window.createOutputChannel("Konveyor-Analyzer");
     this.config = vscode.workspace.getConfiguration("konveyor");
+    this.state = state;
+    this.extContext = state.extensionContext;
   }
 
   public start(): void {
     if (!this.canAnalyze()) {
+      this.sendServerStatus(false, "Analysis configuration incomplete.");
       return;
     }
-    exec("java -version", (err) => {
-      if (err) {
-        vscode.window.showErrorMessage("Java is not installed. Please install it to continue.");
-        return;
-      }
-    });
-    exec("mvn -version", (err) => {
-      if (err) {
-        vscode.window.showErrorMessage("Maven is not installed. Please install it to continue.");
+
+    exec("java -version", (javaErr) => {
```

Comment:
future improvement - running those commands take some time - we could run them AFTER the server failed and/or run them once (and store the flag even in `globalState`).